File: core/src/main/java/de/qabel/core/crypto/QblECPublicKey.java
package de.qabel.core.crypto;

import com.google.gson.annotations.JsonAdapter;
import org.spongycastle.util.encoders.Hex;

import java.io.Serializable;
import java.util.Arrays;

/**
 * Elliptic curve public key
 */
@JsonAdapter(QblECPublicKeyIndexTypeAdapter.class)
public class QblECPublicKey implements Serializable {
    public static final int KEY_SIZE_BYTE = 32;
    private byte[] pubKey;

    /**
     * Generate elliptic curve public key from raw byte array
     *
     * @param pubKey Point which represents the public key
     */
    public QblECPublicKey(byte[] pubKey) {
        this.pubKey = pubKey;
    }

    /**
     * Get encoded public key
     *
     * @return public key
     */
    public byte[] getKey() {
        return pubKey;
    }

    public String getReadableKeyIdentifier() {
        return Hex.toHexString(pubKey);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        QblECPublicKey that = (QblECPublicKey) o;

        return Arrays.equals(pubKey, that.pubKey);

    }

    @Override
    public int hashCode() {
        return Arrays.hashCode(pubKey);
    }
}

File: core/src/main/java/de/qabel/core/crypto/QblECPublicKey.kt
package de.qabel.core.crypto

import com.google.gson.annotations.JsonAdapter
import org.spongycastle.util.encoders.Hex

import java.io.Serializable
import java.util.Arrays

/**
 * Elliptic curve public key
 *
 * Generate elliptic curve public key from raw byte array
 *
 * @param pubKey Point which represents the public key
 */
@JsonAdapter(QblECPublicKeyIndexTypeAdapter::class)
class QblECPublicKey (val key: ByteArray) : Serializable {

    val readableKeyIdentifier: String
        get() = Hex.toHexString(key)

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o == null || javaClass != o.javaClass) {
            return false
        }

        val that = o as QblECPublicKey?

        return Arrays.equals(key, that!!.key)

    }

    override fun hashCode(): Int {
        return Arrays.hashCode(key)
    }

    companion object {
        val KEY_SIZE_BYTE = 32
    }
}
